flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/HsSubpartitionFileReaderImpl.java

```diff
@@ -329,13 +327,15 @@ private Optional<BufferIndexOrError> checkAndGetFirstBufferIndexOrError(
         return Optional.ofNullable(peek);
     }
 
-    private void increaseBacklog(Buffer buffer) {
+    @GuardedBy("lock")
+    private void tryIncreaseBacklog(Buffer buffer) {
         if (buffer.isBuffer()) {
             backlog.getAndIncrement();
         }
     }
 
-    private void decreaseBacklog(Buffer buffer) {
+    @GuardedBy("lock")
```

Comment:
Should be removed as `backlog` is not guarded by `lock`.

Reply:
Fixed.